src/main/java/org/apache/commons/io/input/AutoCloseInputStream.java

```diff
@@ -79,6 +83,44 @@ protected void afterRead(final int n) throws IOException {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * AutoCloseInputStream does not support mark/reset no matter what.
+     * 
+     * @see org.apache.commons.io.input.ProxyInputStream#mark(int)
+     */
+    @Override
+    public synchronized void mark(int readlimit) {
+        // Behave as standard InputStream not supporting mark/reset
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * AutoCloseInputStream does not support mark/reset no matter what.
+     * 
+     * @see org.apache.commons.io.input.ProxyInputStream#reset()
+     */
+    @Override
+    public synchronized void reset() throws IOException {
+        // Behave as standard InputStream not supporting mark/reset
+        throw new IOException("mark/reset not supported");
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * AutoCloseInputStream does not support mark/reset no matter what.
+     * 
+     * @see org.apache.commons.io.input.ProxyInputStream#markSupported()
+     */
+    @Override
+    public boolean markSupported() {
+        // Behave as standard InputStream not supporting mark/reset
+        return false;
+    }
+
```

Comment:
Hi @tmortagne 

Again, this completely changes the contract of the existing class and will break existing call sites. I think there are two kinds of solutions here:
(1) Write a new class as I suggested before.
(2) Implement mark/reset with its own buffer so that reset works even when the underlying stream is actually closed. I'm not even sure of the sanity of that.

Gary

Reply:
I really don't understand this "again". The whole point of the previous conversation was that it was simpler to make AutoCloseInputStream just cancel mark/reset support since it's too complex to really support it properly and you said yourself

> It seems to me like the simplest solution would be for markSupported() to return false.

As I said AutoCloseInputStream does not make much sense in its current state since it break InputStream contract regarding mark/reset support which is not so great. Cancel mark/rest support is one way to prevent this inconsistency which is what I suggested several times and that you seemed to agree on.

If you are suggesting that a new class should be introduced and AutoCloseInputStream deprecated because broken but not fixable it was not very clear.